modules/core/src/convert.cpp

```diff
@@ -4545,6 +4545,16 @@ static short convertFp16SW(float fp32)
 }
 #endif
 
+#if CV_FP16 && (defined __GNUC__) && (defined __arm__ || defined __aarch64__)
+    #if 5 <= __GNUC__
+    inline float16x4_t load_f16(const short* p) { return vld1_f16((const float16_t*)p); }
+    inline void store_f16(short* p, float16x4_t v) { vst1_f16((float16_t*)p, v); }
+    #else
+    inline float16x4_t load_f16(const short* p) { return (float16x4_t)vld1_s16(p); }
+    inline void store_f16(short* p, float16x4_t v) { vst1_s16(p, (int16x4_t)v); }
```

Comment:
Casting between int16x4_t <-> float16x4_t looks dangerous to me.
I believe "reinterpret_cast" should be used in this case to prevent any real data conversion. Does it work here?